rainier-core/src/main/scala/com/stripe/rainier/core/RandomVariable.scala

```diff
@@ -80,6 +81,17 @@ class RandomVariable[+T](val value: T, val targets: Set[Target]) {
       }
   }
 
+  def waic(sampler: Sampler,
```

Comment:
we have too many variations of sampling methods that have these same basic parameters, this is crying out for a refactoring

Reply:
Agreed. I think having `waic` live on some `Sample` type would be much better. 

Meanwhile, should there be a version of the `waic` method that takes a list of samples? I imagine it'll be pretty common to both want the sample from the posterior and the WAIC.

Reply:
> a version of the waic method that takes a list of samples

The problem is that what the waic method needs is the samples of the raw parameter valus, whereas when you `sample()` you're getting some transformed output/prediction. This is why I think we need some kind of `Sample` or `Posterior` object that encapuslates both...